lib/agent.js

```diff
@@ -110,12 +110,14 @@ Agent.prototype.startSpan = function (name, type, subtype, action, { childOf } =
 Agent.prototype._config = function (opts) {
   this._conf = config(opts)
 
-  const { host, port } = this._conf.serverUrl
+  const { host, port, protocol } = this._conf.serverUrl
     ? parsers.parseUrl(this._conf.serverUrl)
     : { host: 'localhost:8200', port: '8200' }
 
   this._conf.serverHost = host
-  this._conf.serverPort = parseInt(port, 10)
+  this._conf.serverPort = port === ''
+    ? (protocol === 'https:' ? 443 : 80)
```

Comment:
I don't suspect we need to worry about a protocol other than "https" or "http" here. :)